src/drive/components/FilesViewer.jsx

```diff
@@ -35,27 +43,77 @@ class FilesViewer extends Component {
     this.fetchMore()
   }
 
-  render() {
-    if (this.props.files.length === 0) return null
-    // TODO: temp fix for thumbnail links on mobile
-    if (isCordova() && !this.props.filesWithLinks) return null
-    const files = isCordova()
-      ? this.props.filesWithLinks
+  needLinks() {
+    return isCordova() && this.state.loading
+  }
+
+  getFiles() {
+    return isCordova()
```

Comment:
Again I think `isCordova` is the wrong condition here. If we're using cordova but all files are PDFs, we don't want to fetch the links either !